components/com_joomgallery/views/detail/tmpl/default.php

```diff
@@ -197,7 +197,31 @@
         <div style="white-space:nowrap;" id="trueContainer">
 <?php   endif;
         if(count($this->images)):
-          foreach($this->images as $row): ?>
+
+			$image	= $this->get('Image');
+			$rows		= [];
+			$limit	= $this->_config->get('jg_motionminiLimit');
+			if ($limit != 0)
+			{
+				$limit	= (int)(($limit - 1) / 2);
+
+				for($i = -$limit; $i <= $limit; $i++)
+				{
+					$imageposition = $image->position + $i;
+
+					if ( $imageposition < 0 or $imageposition >= count($this->images) )
+					{
+						continue;
+					}
+					$rows[] = $this->images[$imageposition];
+				}
+			}
+			else
+			{
+				$rows = $this->images;
+			}
```

Comment:
I would try to avoid using a for loop here because with array methods like array_splice or array_slice you should be able to achieve the same.
For example try to replace the complete block of code with this:

```php
$limit = $this->_config->get('jg_motionminiLimit');
if ($limit > 0)
{
  $rows = array_splice($this->images, max(0, $this->image->position - $limit/2), $limit);
}
else
{
  $rows = $this->images;
}
```

I haven't tested this but something like this should work (the `max` method ensures that we don't end up with a negative index).